File: backend/app/schemas/notification.py
"""通知相关 Schema"""
from pydantic import BaseModel
from typing import Optional, List, Dict, Any
from datetime import datetime


class SenderInfo(BaseModel):
    id: str
    username: str

    class Config:
        from_attributes = True


class NotificationResponse(BaseModel):
    id: str
    notification_type: str
    title: str
    content: str
    is_read: bool
    status: str
    data: Dict[str, Any]
    sender: Optional[SenderInfo] = None
    created_at: datetime

    class Config:
        from_attributes = True


class NotificationListResponse(BaseModel):
    items: List[NotificationResponse]
    total: int
    page: int
    size: int
    has_next: bool
    unread_count: int


class SystemAnnouncementCreate(BaseModel):
    title: str
    content: str

File: frontend/src/types/notification.ts
export interface NotificationSender {
  id: string
  username: string
}

export interface Notification {
  id: string
  notification_type: 'project_invite' | 'system_announcement' | 'community_interaction' | 'analysis_complete'
  title: string
  content: string
  is_read: boolean
  status: 'pending' | 'accepted' | 'rejected' | 'done'
  data: Record<string, any>
  sender?: NotificationSender
  created_at: string
}

export interface NotificationListResponse {
  items: Notification[]
  total: number
  page: number
  size: number
  has_next: boolean
  unread_count: number
}
